models/DataObject/ClassDefinition/Data/CalculatedValue.php

```diff
@@ -338,12 +344,32 @@ public function getGetterCodeLocalizedfields($class)
         $code .= "\t\t" . '}' . "\n";
         $code .= "\t" . '}'  . "\n";
 
+        if($class instanceof DataObject\Objectbrick\Definition) {
+            $ownerType = 'objectbrick';
+            $ownerName = $class->getClassDefinitions()[0]['fieldname']; // todo this is not really correct. Actually we need to find out the name of the brick container (an instance of \Pimcore\Model\DataObject\Objectbrick). This must happen during runtime as one brick can be used in multiple brick containers.
```

Comment:
what is the right interpretation of the comment ?

Reply:
This means that we cannot determine the name of the class with the brick container field when the brick gets saved. Actually we need to add some code in the generated getter function to determine the brick container field during runtime but I did not find a way to do that because the generated brick class does not know anything in which brick container / data object class it is being used.
And because of that I chose a way to at least enhance the status quo (although this solution is not perfect) by setting the `$ownerName` to the first configured brick container field of the object brick.

Reply:
Hi, just noticed something. 

https://github.com/pimcore/pimcore/blob/2598b5dc343e06f5100cc8ce8d90c33c6301baf8/models/DataObject/ClassDefinition/Data/Localizedfields.php#L142

The calculator only gets called if there are other fields with data in the localized field container?

What do you think?

Reply:
Why not just take it from $this ?

![image](https://user-images.githubusercontent.com/6142086/71252030-2e409280-2324-11ea-9699-021a4accfe82.png)

Reply:
Is `fieldname` the calculated value field in the brick or the brick container field in the data object class? I need the latter.

Reply:
the one in the data object class

Reply:
Oh, then have overseen this. Thanks, I will change it.

Reply:
Done. My test was positive.

Reply:
Can you also check this "I noticed something" thing ?